hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/NelsonCondition.java

```diff
@@ -0,0 +1,244 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.api.model.condition;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hawkular.alerts.api.model.data.Data;
+import org.hawkular.alerts.api.model.trigger.Mode;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * A condition that uses historically collected data to perform a variety of tests for value instability. See:
+ * https://en.wikipedia.org/wiki/Nelson_rules.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@ApiModel(description = "A condition to detect instability based on historical data. + \n" +
+        " + \n" +
+        "From one to all of the defined Nelson rules can be evaluated. See + \n" +
+        "https://en.wikipedia.org/wiki/Nelson_rules for a description of the rules.")
+public class NelsonCondition extends Condition {
+    private static final long serialVersionUID = 1L;
+
+    public enum NelsonRule {
+        Rule1("One point is more than 3 standard deviations from the mean"), //
+        Rule2("Nine (or more) points in a row are on the same side of the mean."), //
+        Rule3("Six (or more) points in a row are continually increasing (or decreasing)."), //
+        Rule4("Fourteen (or more) points in a row alternate in direction, increasing then decreasing."), //
+        Rule5("At least 2 of 3 points in a row are > 2 standard deviations from the mean in the same direction."), //
+        Rule6("At least 4 of 5 points in a row are > 1 standard deviation from the mean in the same direction."), //
+        Rule7("Fifteen points in a row are all within 1 standard deviation of the mean on either side of the mean."), //
+        Rule8("Eight points in a row exist, but none within 1 standard deviation of the mean, "
+                + "and the points are in both directions from the mean.");
+
+        private String description;
+
+        NelsonRule(String description) {
+            this.description = description;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+    }
+
+    private static final int DEFAULT_SAMPLE_SIZE = 50;
+    private static final Set<NelsonRule> DEFAULT_ACTIVE_RULES = EnumSet.allOf(NelsonRule.class);
+
+    @JsonInclude(Include.NON_NULL)
+    private String dataId;
+
+    @ApiModelProperty(value = "Set of NelsonRule to evaluate. Default = All Rules.", position = 1, required = false)
+    @JsonInclude(Include.NON_NULL)
+    private Set<NelsonRule> activeRules;
```

Comment:
I see activeRules field has been mapped as a set<text> in cassandra.
This can not be null as C* could throw unexpected errors on that, perhaps some IllegalArgumentException if some null on constructor or setter would be good.